### DISCUSSION TITLE
.NET 6 DateOnly and TimeOnly support

### DISCUSSION THREAD

**Discussion by ryanbuening** [ravendb/ravendb#12797] [Q&A] [answered]
Hello. I was just wondering if RavenDB supports or will support the new [DateOnly and TimeOnly types in .NET 6](https://devblogs.microsoft.com/dotnet/date-time-and-time-zone-enhancements-in-net-6/#introducing-the-dateonly-and-timeonly-types)? Thanks!

**Comment by ayende** [comment, marked as answer]
To an extent, yes. The client API may need to update the serialization. We will also not be recognizing them on the server side by default, you'll need to explicitly call `new DateOnly(doc.Birthday)` if you want to deal with the date math in the index.

We'll be upgrading RavenDB to 6.0 at around the 5.3 release, which will allow you to use that in the server side as well.

**Reply by ayende** [reply]
https://issues.hibernatingrhinos.com/issue/RavenDB-17250

**Reply by ryanbuening** [reply]
Sorry, what do you mean by the "client API may need to update the serialization"? Are you referring to using a converter?

Any ballpark estimate on when 5.3/6.0 will be released? Thanks.

**Reply by ayende** [reply]
I'm talking about converter, yes. And we'll likely release 5.3 a few weeks after the .NET 6.0 release

**Reply by ryanbuening** [reply]
Got it. And then with 5.3, a custom converter wouldn't be necessary as the types would be supported out of the box?

**Reply by ayende** [reply]
Correct

**Reply by ryanbuening** [reply]
I was reading the [changelog for 5.3](https://ravendb.net/docs/article-page/5.3/Csharp/start/whats-new#53000), but didn't see anything about DateOnly or TimeOnly. Are these changes in there?

**Reply by ppekrol** [reply]
We will be bumping RavenDB 5.2+ to .NET 6 in early 2022, we will add a support for those types afterwards.
